src/payment/strategies/paypal-commerce/paypal-commerce-hosted-form.ts

```diff
@@ -0,0 +1,227 @@
+import { isNil, kebabCase, omitBy } from 'lodash';
+
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+
+import { PaypalCommerceFormFieldStyles, PaypalCommerceFormFieldStylesMap, PaypalCommerceFormFieldType, PaypalCommerceFormFieldValidateEventData, PaypalCommerceFormOptions, PaypalCommerceHostedFields, PaypalCommerceHostedFieldsRenderOptions, PaypalCommerceHostedFieldsState, PaypalCommerceRegularField, PaypalCommerceRequestSender, PaypalCommerceSDK } from './index';
+import { PaypalCommerceFormFieldsMap, PaypalCommerceStoredCardFieldsMap } from './paypal-commerce-payment-initialize-options';
+
+enum PaypalCommerceHostedFormType {
+    CreditCard,
+    StoredCardVerification,
+}
+
+export default class PaypalCommerceHostedForm {
+    private _formOptions?: PaypalCommerceFormOptions;
+    private _cardNameField?: PaypalCommerceRegularField;
+    private _hostedFields?: PaypalCommerceHostedFields;
+    private _type?: PaypalCommerceHostedFormType;
+
+    constructor(
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender
+    ) {}
+
+    async initialize(options: PaypalCommerceFormOptions, cartId: string, paypal: PaypalCommerceSDK) {
+        if (!paypal.HostedFields) {
+            return false;
+        }
+
+        this._formOptions = options;
+        this._type = this._isPaypalCommerceFormFieldsMap(options.fields) ?
+            PaypalCommerceHostedFormType.CreditCard :
+            PaypalCommerceHostedFormType.StoredCardVerification;
+
+        if (paypal.HostedFields.isEligible()) {
```

Comment:
What happens if it's not eligible? Do we just want to fail silently? What does the experience look like for end users?

Reply:
Good question. We should throw error. I'll add it
Thank you!